### PR TITLE
Changing to Array to save un-needed space occupied by WrappedArray

### PR DESCRIPTION
**Pull request by pankajgupta**
Hey @szymonm -- will appreciate if you can take a quick look.

### CODE REVIEW

**Comment by szymonm** [comment]
IIUC this saves creating 2 wrapped arrays per node, but even after this change we create them after calling for neighbors that returns Seq, right?

**Comment by pankajgupta** [comment]
Yes, it's not stored per node though

File: cassovary-core/src/main/scala/com/twitter/cassovary/util/io/ListOfEdgesGraphReader.scala [hunk 1]
```diff
@@ -112,7 +112,7 @@ class ListOfEdgesGraphReader[T](
 
       val edgesBySource = readEdgesBySource()
 
-      private lazy val edgesIterator = edgesBySource.entrySet().iterator()
+      private lazy val edgesIterator = edgesBySource.int2ObjectEntrySet().fastIterator()
```
Comment (szymonm) [inline cassovary-core/src/main/scala/com/twitter/cassovary/util/io/ListOfEdgesGraphReader.scala]: Did you checked what we gain by using fastIterator?

File: cassovary-core/src/main/scala/com/twitter/cassovary/graph/node/SharedArrayBasedDirectedNode.scala [hunk 2]
```diff
@@ -36,7 +36,7 @@ object SharedArrayBasedDirectedNode {
     dir match {
       case StoredGraphDir.OnlyIn | StoredGraphDir.OnlyOut | StoredGraphDir.Mutual =>
         val outEdges: Seq[Int] = Option(edges(nodeId)).getOrElse(emptyArray)
-        UniDirectionalNode(nodeId, outEdges, dir)
+        UniDirectionalNode.applySeq(nodeId, outEdges, dir)
```
Comment (szymonm) [inline cassovary-core/src/main/scala/com/twitter/cassovary/graph/node/SharedArrayBasedDirectedNode.scala]: We should have here outEdges: Array[Int] as well. But this id something for next commits. And let's wait for it until I finish my collections refactoring.

**Comment by szymonm** [comment]
looks good to me

File: cassovary-core/src/main/scala/com/twitter/cassovary/util/io/ListOfEdgesGraphReader.scala (hunk 1 above)
Comment (pankajgupta) [inline cassovary-core/src/main/scala/com/twitter/cassovary/util/io/ListOfEdgesGraphReader.scala]: No, I didn't do any benchmarks but is recommended by documentation on fastutils.

File: cassovary-core/src/main/scala/com/twitter/cassovary/graph/node/SharedArrayBasedDirectedNode.scala (hunk 2 above)
Comment (pankajgupta) [inline cassovary-core/src/main/scala/com/twitter/cassovary/graph/node/SharedArrayBasedDirectedNode.scala]: ok, doing nothing now then.

**Pull request merged**